### ISSUE TITLE
New Variables: Cascading menus for variable references

### ISSUE THREAD

**Issue by vicchig** [MajeedKazemi/nova-editor#238]
New variable reference buttons should display a cascading menu of all possible options for actions that can be performed with the variable reference at a given position when you hover over their reference buttons in the toolbox. 

**On Empty Line/On Empty Hole**
- **Numbers:**
    - None because we can only reference the variable on an empty line
- **Booleans:**
    - None because we can only reference the variable on an empty line
- **Strings:** 
    - None because we can only reference the variable on an empty line
- **Lists:**
    - None because we can only reference the variable on an empty line

**After the Identifier of that Variable (On an empty line)**
- **Numbers:**
    - +=
    - -=
    - *=
    - /=
    - =
- **Booleans:**
    - =
- **Strings:** 
    - +=
    - =
    - Available string methods (void or otherwise)
- **Lists:**
    - =
    - [---]
    - Available list methods (void or otherwise)

**After the Identifier of that Variable (On an empty hole)**
- **Numbers:**
    - Comparison operators
    - Arithmetic operators
    - Basically all options that our current implementation allows in this context
- **Booleans:**
    - Boolean operators (unary and binary)
- **Strings:** 
    - Available string methods (non-void)
- **Lists:**
    - [---]
    - Available list methods (non-void)

**Options that would send the code into draft mode upon insertion should be highlighted as well in the cascaded menu.**

**Comment by vicchig** [comment]
Need to create all the different operators first, then we can build the cascading menu using a single level version of the current SuggestionMenu. This should not be hard, but it will likely require updating the maps in Util.ts to contain the new constructs.
